### ISSUE TITLE
Using Automator/Radarr/Deluge on separate VMs 

### ISSUE THREAD

**Issue by tscibilia** [mdhiggins/sickbeard_mp4_automator#887]
I've been using Radarr, Deluge on separate VMs for a while now and would like to automate the post-processing using the scripts you've developed here. It seems to be intended to be used on one machine, so I'm having difficulty getting each to talk to one another. I made an attempt at bridging the gap... 

**Radarr run a custom script:**
`ssh -i ~/.ssh/sshkey-rsa root@192.168.5.123 "~/sickbeard_mp4_automator/postRadarr.py"`
**and Deluge to run this script:**
`ssh -i ~/.ssh/sshkey-rsa root@192.168.5.123 "~/sickbeard_mp4_automator/delugePostProcess.py"`

My initial thought is this will fail because there are no variables being passed to the python script but I decided to run the commands manually in order to troubleshoot. 

**When I ran the script Deluge would process upon completion, I received this error:**
```
root@deluge:~# /home/vpn/postDnld.sh
Traceback (most recent call last):
  File "/root/sickbeard_mp4_automator/delugePostProcess.py", line 8, in <module>
    from deluge_client import DelugeRPCClient
ImportError: No module named 'deluge_client'
```
**When I ran the script Radarr would process upon completion, I received this error:**
```
root@media:~# /home/radarr/PostDnld.sh 
2018-06-07 17:03:30 - RadarrPostConversion - INFO - Radarr extra script post processing started.
2018-06-07 17:03:30 - readSettings - INFO - /usr/bin/python
2018-06-07 17:03:30 - readSettings - WARNING - Audio bitrate was invalid, defaulting to 256 per channel.
2018-06-07 17:03:30 - RadarrPostConversion - INFO - environ({'XDG_RUNTIME_DIR': '/run/user/0', 'SHLVL': '1', 'L
ANG': 'C', 'LOGNAME': 'root', 'USER': 'root', 'PWD': '/root', 'MAIL': '/var/mail/root', 'SHELL': '/bin/bash', '
XDG_SESSION_ID': '13953', 'SSH_CLIENT': '192.168.5.112 38372 22', 'PATH': '/usr/local/sbin:/usr/local/bin:/usr/
sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games', 'SSH_CONNECTION': '192.168.5.112 38372 192.168.5.123 22'
, 'HOME': '/root', '_': '/root/sickbeard_mp4_automator/postRadarr.py'})
Traceback (most recent call last):
  File "/root/sickbeard_mp4_automator/postRadarr.py", line 33, in <module>
    if MkvtoMp4(settings).validSource(inputfile):
  File "/root/sickbeard_mp4_automator/mkvtomp4.py", line 243, in validSource
    input_dir, filename, input_extension = self.parseFile(inputfile)
  File "/root/sickbeard_mp4_automator/mkvtomp4.py", line 761, in parseFile
    path = os.path.abspath(path)
  File "/usr/lib/python3.5/posixpath.py", line 357, in abspath
    if not isabs(path):
  File "/usr/lib/python3.5/posixpath.py", line 64, in isabs
    return s.startswith(sep)
AttributeError: 'NoneType' object has no attribute 'startswith'
```

Any ideas on how I can get this setup to work?

**Comment by mdhiggins** [comment]
The deluge error means you need to install the deluge_client module to proceed, that's an easy one, just use pip; see readme

The Radarr script needs to be executed by Radarr and have all its environment variables that Radarr injects to pass on the relevant data so without those that won't work either

Reference:
https://github.com/Radarr/Radarr/wiki/Custom-Post-Processing-Scripts

Without know more about your setup this would be very hard to offer advice

**Comment by tscibilia** [comment]
thanks, I looked over the readme again and missed the pip deluge_client. I installed it and reran my script and received a promising error back (I assume this will work when run from deluge).

`2018-06-09 16:30:57 - delugePostProcess - INFO - Deluge post processing started.`
`2018-06-09 16:30:57 - delugePostProcess - ERROR - Not enough command line parameters present, are you launching this from deluge?`

**I'll try letting this run through and execute via deluge and radarr and report back.** BTW, let me know what info you'd need from me about my setup.

**Comment by mdhiggins** [comment]
Yeah both of those scripts should work if run from their respective locations
Try from deluge/radar and see

As far what info I need, I guess just how you structure your storage and how the files are passed from one VM to another

**Comment by tscibilia** [comment]
I realized I had some additional settings incorrect. namely, the scripts I created to remotely trigger the deluge and radarr python scripts were being run as the user I setup for each application (not root) so I adjusted them to have the right permissions and paths. 

**I thought the deluge script ran without error but after a while returns this msg:**
`[WARNING ] 01:36:47 core:132 [execute] command '/home/vpn/postDnld.sh' failed with exit code 255`
`[WARNING ] 01:36:47 core:136 [execute] stderr: ssh: connect to host 192.168.5.123 port 22: Connection timed out`

**However the radarr script still has trouble running, I can't confirm it runs** 
	--When I trigger it manually I get the same message I posted earlier. 
	--When I let a download complete, nothing shows up in the radarr logs.

**As for my setup/configuration:**
 - I'm running proxmox with separate lxc containers for each VM (setup using instructions by [htpcguides.com](https://www.htpcguides.com)
 - VMs run on the server local disk, each has modified fstab file to mount network folders to access media
 - proxmox server v4.4 (all VMs on Ubuntu 16.04.4 & none of the services are run as root)
  -- (deluge-vm) setup as split tunnel with OpenVPN [htpcguides setup](https://www.htpcguides.com/configure-deluge-for-vpn-split-tunneling-ubuntu-16-04/)
  -- (media-vm) sonarr and radarr on same VM [htpcguides setup](https://www.htpcguides.com/install-nzbdrone-ubuntu/)
  -- (plex-vm) self explainatory
  -- (ffmpeg-vm) higher powered VM running ffmpeg & filebot & sb_mp4_automator for transcoding and handling downloaded vids

In summary, I'm not sure how to make this work, I suppose I could try to combine all these services into one VM (but I never liked the idea of all the eggs in one basket). Let me know if anything sticks out to you or if you came across someone who might have run into a similar setup that I could reach out to.

**Comment by tscibilia** [comment]
@mdhiggins I'm going to close this issue, I never resolved connecting the automation script across different VMs. But in the interest of getting this to benefit my workflow, I just combined the applications managing my media into one VM and it seems to trigger the scripts. 

Now I just need help tweaking it for my workflow (so I'll open another issue for that).
